File: keprix_sdk/python/src/keprix_sdk/schema.py
from __future__ import annotations

from typing import Any

from keprix_sdk.domain import Domain


def domain_to_json(domain: Domain) -> dict[str, Any]:
    return {
        "name": domain.name,
        "entities": [
            {
                "name": entity.name,
                "fields": [
                    {
                        "name": field.name,
                        "type": field.type,
                        "required": field.required,
                        "default": field.default,
                        "entity": field.entity,
                        "values": field.values,
                    }
                    for field in entity.fields
                ],
                "operations": [
                    {
                        "name": operation.name,
                        "confirmation_required": operation.confirmation_required,
                    }
                    for operation in entity.operations
                ],
            }
            for entity in domain.entities
        ],
    }

File: keprix_sdk/typescript/src/schema.ts
import type { Domain } from "./domain.js";

export function domainToJson(domain: Domain): Record<string, unknown> {
  return {
    name: domain.name,
    entities: (domain.entities || []).map((entity) => ({
      name: entity.name,
      fields: (entity.fields || []).map((field) => ({
        name: field.name,
        type: field.type ?? "string",
        required: field.required ?? false,
        default: field.default,
        entity: field.entity,
        values: field.values,
      })),
      operations: (entity.operations || []).map((operation) => ({
        name: operation.name,
        confirmation_required: operation.confirmation_required ?? false,
      })),
    })),
  };
}
